kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/sink/config/S3SinkConfigDef.scala

```diff
@@ -288,8 +133,21 @@ class S3ConfigDef() extends ConfigDef with LazyLogging {
 
 }
 
-case class S3ConfigDefBuilder(sinkName: Option[String], props: util.Map[String, String])
-    extends BaseConfig(S3ConfigSettings.CONNECTOR_PREFIX, S3ConfigDef.config, props)
+object S3SinkConfigDefBuilder {
+  def apply(
+    props:        util.Map[String, String],
+    contextProps: util.Map[String, String],
+  ): S3SinkConfigDefBuilder =
+    S3SinkConfigDefBuilder(
+      Option(contextProps)
+        .filterNot(_.isEmpty)
+        .getOrElse(props),
```

Comment:
can `props` be null? 

Minor: it would help to name them props + fallback/default. This might instruct what this does.

Reply:
Refactored to make this behaviour clearer and further protect against nulls.